Image `argoproj/argocd-test-tools` running an old node version

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Running `make dep-ui` after pulling the `argoproj/argocd-test-tools` resolves in an error:
`foundation-sites@6.7.4: The engine "node" is incompatible with this module. Expected version ">=12.0". Got "11.15.0"`

 I pulled the `argoproj/argocd-test-tools` and then run `make dep-ui`

**To Reproduce**
Following the [toolchain-guide](https://argo-cd.readthedocs.io/en/stable/developer-guide/toolchain-guide/#build-or-pull-the-required-docker-image)
1. `docker pull argoproj/argocd-test-tools`
2. `export TEST_TOOLS_NAMESPACE=argoproj`
3. `export SKIP_TEST_TOOLS_IMAGE=1`
4. `make dep-ui`

**Expected behavior**

Complete the step successfully 

**Logs**

```
docker run --rm -it --name argocd-test-client -u 501:20 -e HOME=/home/user -e GOPATH=/go -e ARGOCD_E2E_K3S= -e GITHUB_TOKEN -e GOCACHE=/tmp/go-build-cache -e ARGOCD_LINT_GOGC=20 -v "/Users/omer/dev/argo-cd:/go/src/github.com/argoproj/argo-cd:delegated" -v /Users/omer/go/pkg/mod:/go/pkg/mod:delegated -v /Users/omer/.cache/go-build:/tmp/go-build-cache:delegated -v /Users/omer/.kube:/home/user/.kube:delegated -v /tmp:/tmp:delegated -w /go/src/github.com/argoproj/argo-cd argoproj/argocd-test-tools:latest bash -c "make dep-ui-local"
cd ui && yarn install
yarn install v1.15.2
[1/4] Resolving packages...
[2/4] Fetching packages...
error foundation-sites@6.7.4: The engine "node" is incompatible with this module. Expected version ">=12.0". Got "11.15.0"
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
make: *** [Makefile:571: dep-ui-local] Error 1
make: *** [dep-ui] Error 2
```

--- a/docs/developer-guide/toolchain-guide.md
+++ b/docs/developer-guide/toolchain-guide.md
@@ -157,9 +157,9 @@ Make sure you fulfill the pre-requisites above and run some preliminary tests. N
 * Run `docker version`
 * Run `go version`
 
-### Build (or pull) the required Docker image
+### Build the required Docker image
 
-Build the required Docker image by running `make test-tools-image` or pull the latest version by issuing `docker pull argoproj/argocd-test-tools`.
+Build the required Docker image by running `make test-tools-image`. This image offers the environment of the virtualized toolchain.
 
 The `Dockerfile` used to build these images can be found at `test/container/Dockerfile`.
 

